torch/tensor.py

```diff
@@ -504,10 +504,13 @@ def __cuda_array_interface__(self):
         itemsize = self.storage().element_size()
 
         shape = tuple(self.shape)
-        strides = tuple(s * itemsize for s in self.stride())
+        if self.is_contiguous():
+            strides = None
+        else:
+            strides = tuple(s * itemsize for s in self.stride())
```

Comment:
according to the protocol, we need to zero out data here if numel is 0.

Reply:
I don get it. What's wrong here? It looks fine to me.

Reply:
the line below should be:
`data = (self.data_ptr() if self.numel() > 0 else 0, False)  # read-only is false`

Reply:
Oh, good catch @gchanan! (Knew nothing about `numel()`, just looked it up from the doc.) In fact I also caught this myself for CuPy just yesterday...

Reply:
Thanks, I fixed it.